paddle/fluid/operators/affine_grid_op.h

```diff
@@ -37,29 +37,33 @@ using Array4 = Eigen::DSizes<int64_t, 4>;
  */
 template <typename DeviceContext, typename T>
 struct Linspace {
-  void operator()(T start, T end, int count, framework::Tensor* numbers,
+  void operator()(T start, T end, int count, bool align_corners,
+                  framework::Tensor* numbers,
                   const framework::ExecutionContext& ctx);
 };
 
 template <typename DeviceContext, typename T>
-inline void GetIdxMap(int n, int h, int w, Tensor* grid,
+inline void GetIdxMap(int n, int h, int w, bool align_corners, Tensor* grid,
                       const framework::ExecutionContext& ctx) {
   auto& place = *ctx.template device_context<DeviceContext>().eigen_device();
   grid->mutable_data<T>({n, h, w, 3}, ctx.GetPlace());
   auto grid_t = EigenTensor<T, 4>::From(*grid);
   // Get indexes of height with shape [height, width, 1]
   Tensor h_idx;
   Linspace<DeviceContext, T> linspace;
-  linspace((T)-1, (T)1, h, &h_idx, ctx);
+  linspace((T)-1, (T)1, h, align_corners, &h_idx, ctx);
```

Comment:
`linespace` seems to be simple elementwise computation, why isn't eigen used for it too?

Reply:
Some errors occurred in linspace API of eigen  when the data type is float64. I haven't found the reason currently.
And it is unrelated to performance on GPU devices after CUDA kernel was added. So may I fix it in the next PR?

Reply:
yes, that would be fine.